### PR TITLE
#924 [Win32] Possible ERROR_GRAPHIC_DISPOSED error during painting

### PR DESCRIPTION
**Pull request by tmssngr**
This should cover the case that BeginBufferedPaint might fail.

### CODE REVIEW

**Comment by tmssngr** [comment]
Where can I find what check failed?

**Comment by tmssngr** [comment]
Can somebody please merge this request? I don't think that the failing GTK tests are caused by this patch.

**Comment by BeckerWdf** [comment]
So this PR fixes https://github.com/eclipse-platform/eclipse.platform.swt/issues/924 ?

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java [hunk 1]
```diff
@@ -1521,30 +1521,32 @@ LRESULT WM_PAINT (long wParam, long lParam) {
...
-				data.font = Font.win32_new(display, OS.SendMessage (handle, OS.WM_GETFONT, 0, 0), nativeZoom);
-				data.uiState = (int)OS.SendMessage (handle, OS.WM_QUERYUISTATE, 0, 0);
-				if ((style & SWT.NO_BACKGROUND) != 0) {
-					/* This code is intentionally commented because it may be slow to copy bits from the screen */
-					//paintGC.copyArea (image, ps.left, ps.top);
-				} else {
-					RECT rect = new RECT ();
-					OS.SetRect (rect, ps.left, ps.top, ps.right, ps.bottom);
-					drawBackground (phdc [0], rect);
+				if (hBufferedPaint != 0 && phdc[0] != 0) {
```
Comment (BeckerWdf) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java:1524]: The diff looks a bit complicated. Do I get it right that the only change is to add the

if (hBufferedPaint != 0 && phdc[0] != 0)

condition?

Comment (tmssngr) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java:1524]: Yes.

**Comment by BeckerWdf** [comment]
I am not an expert on that topic and even do no work on windows. So somebody else should  have a look at it.

**Comment by HeikoKlare** [comment]
In general, the change looks sound to me. I am just wondering whether it is necessary and thus whether we should merge this change now. We are already past M3 and even though I would consider the risk of the change as low in general, at the same the affected code is quite crucial. The issue does not seem to be critical: the report has been made a year ago and no one else complained so far. So would it be sufficient to target this for 2025-03 M1? Then we have the whole three months of 2025-03 development for implicit testing by developers using Windows.

**Comment by tmssngr** [comment]
Would someone be brave enough to commit to master? Why does the Linux build fail? Shouldn't be caused by this PR.

**Comment by HeikoKlare** [comment]
> Would someone be brave enough to commit to master? Why does the Linux build fail? Shouldn't be caused by this PR.

Yes, I've just rebased the branch onto master. Linux build failures are existing and documented here: #1564
I'm just curious about the reported new Maven issues, which is why I retriggered the Jenkins build.

**Comment by tmssngr** [comment]
Thanks for merging.

**Pull request merged**